### DISCUSSION TITLE
Checked Exception과 Unchecked Exception의 차이

### DISCUSSION THREAD

**Discussion by GIVEN53** [2024-woowacourse-study/level-interview#21] [Java]
\+ 언제 Checked Exception을 쓰고 언제 Unchecked Exception을 써야 할까?
\+ Error와 Exception의 차이

**Comment by kunsanglee** [comment]
Error는 시스템에 비정상적인 상황이 발생한 경우를 말합니다. 주로 JVM에서 발생하는 것 이며, 예외와 반대로 이를 애플리케이션 코드에서 잡을 수 없습니다. 예시로는 OutOfMemoryError, StackOverFlow 등이 있습니다.

Exception은 입력값에 대한 처리가 불가능하거나, 프로그램 실행 중에 참조값이 잘못 된 경우 등 정상적인 프로그램 흐름을 어긋나는 경우를 말합니다. 자바에서는 예외를 개발자가 미리 예측하여 직접 핸들링 할 수 있습니다.

또한 예외는 CheckedException과 UncheckedException으로 구분할 수 있습니다. 
CheckedException은 RuntimException을 상속받지 않은 클래스입니다. 컴파일 시점에 확인할 수 있으며 명시적인 예외 처리를 해야합니다.
UncheckedException은 RuntimeException을 상속받은 클래스입니다. 런타임 시점에 확인할 수 있으며 명시적인 예외 처리를 하지 않습니다.

**Comment by seunghye218** [comment]
Checked Exception은 애플리케이션이 예상하고 복구해야 하는 예외입니다. 이러한 예외는 개발자가 반드시 catch 하거나 메서드 시그니처에서 선언하여 처리해야 합니다.  
예외가 처리되거나 명시적으로 선언되도록 하려면 Checked Exception를 사용해야 합니다. 

Unchecked Exception은 애플리케이션의 결함이나 애플리케이션이 일반적으로 예상하거나 복구할 수 없는 상황을 나타내는 예외입니다. 이 예외는 메서드 시그니처에서 포착하거나 선언할 필요가 없습니다. 

Error와 Exception의 차이점으로 error는 일반적으로 복구할 수 없는 심각한 문제를 나타내는 반면, Exception는 애플리케이션 코드 내에서 잠재적으로 처리 및 복구할 수 있는 조건을 나타냅니다.
